qa/full-cluster-restart/src/test/java/org/elasticsearch/upgrades/FullClusterRestartIT.java

```diff
@@ -1132,7 +1121,7 @@ private void checkSnapshot(String snapshotName, int count, Version tookOnVersion
 
         // In 7.0, type names are no longer returned by default in get index template requests.
         // We therefore use the deprecated typed APIs when running against the current version.
-        if (isRunningAgainstOldCluster() == false) {
+        if (isRunningAgainstAncientCluster() == false) {
```

Comment:
Based on the behaviour of `isRunningAgainstOldCluster` (but see my comment there about the name vs behaviour), this check seems wrong.

If the old cluster is 7.0.0 then `isRunningAgainstAncientCluster` will be false, and `INCLUDE_TYPE_NAME_PARAMETER` will be true, but I think we only want that if old version is pre-7 and we're running on the new version.

Similarly if we're testing against `6.7.x` the parameter will be added regardless of which cluster we're running against (old or new), which is not how this test worked before.

But maybe I'm confused about why we want this parameter.

```suggestion
        if (isRunningAgainstOldCluster() == false && getOldClusterVersion().major < 7) {
```